bin/set_allowed_mins.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+
+"""
+Handles updating the database with allowed minutes
+and override allowed minutes for the gradeable timer
+"""
+
+from sqlalchemy import create_engine, MetaData, text
+import datetime
+import os
+import sys
+import json
+import re
+
+try:
+    CONFIG_PATH = os.path.join(
+        os.path.dirname(os.path.realpath(__file__)), '..', 'config')
+
+    with open(os.path.join(CONFIG_PATH, 'submitty.json')) as open_file:
+        SUBMITTY_CONFIG = json.load(open_file)
+
+    with open(os.path.join(CONFIG_PATH, 'database.json')) as open_file:
+        DATABASE_CONFIG = json.load(open_file)
+
+except Exception as config_fail_error:
+    print("[{}] ERROR: CORE SUBMITTY CONFIGURATION ERROR {}".format(
+        str(datetime.datetime.now()), str(config_fail_error)))
+    sys.exit(1)
+
+DB_HOST = DATABASE_CONFIG['database_host']
+DB_USER = DATABASE_CONFIG['database_user']
+DB_PASSWORD = DATABASE_CONFIG['database_password']
+
+CONFIG_FILE_PATH = sys.argv[1]
+SEMESTER = sys.argv[2]
+COURSE = sys.argv[3]
+GRADEABLE = sys.argv[4]
+
+
+def setup_db():
+    """Set up a connection with the course database."""
+    db_name = "submitty_{}_{}".format(SEMESTER, COURSE)
+    # If using a UNIX socket, have to specify a slightly different connection string
+    if os.path.isdir(DB_HOST):
+        conn_string = "postgresql://{}:{}@/{}?host={}".format(
+            DB_USER, DB_PASSWORD, db_name, DB_HOST)
+    else:
+        conn_string = "postgresql://{}:{}@{}/{}".format(
+            DB_USER, DB_PASSWORD, DB_HOST, db_name)
+
+    engine = create_engine(conn_string)
+    db = engine.connect()
+    metadata = MetaData(bind=db)
+    return db, metadata
+
+
+def send_data(db, allowed_minutes, override):
+    query = """UPDATE gradeable SET g_allowed_minutes = :minutes
+               WHERE g_id=:gradeable"""
+    db.execute(text(query), minutes=allowed_minutes, gradeable=GRADEABLE)
+    query = """DELETE FROM gradeable_allowed_minutes_override WHERE g_id=:gradeable"""
+    db.execute(text(query), gradeable=GRADEABLE)
+    if override is not None:
+        for user in override:
```

Comment:
What if the username is not a valid user in the course?  It appears we blindly add them to the database.  So it doesn't crash or error out -- that's good...  but it would be better if the database had a key constraint (please lets add one) on the user_id column to be a valid user.  And then this chunk of code would need to handle the error of a non existent user in the config file.  It would be nice if this error got spit back out to the build script STDOUT/STDERR -- whatever we are capturing for the webpage log.

<img width="373" alt="Screen Shot 2021-07-23 at 1 04 32 AM" src="https://user-images.githubusercontent.com/5871417/126739418-115d6aa6-032b-41a7-b572-5f870cacef7d.png">